src/state/container/mod.rs

```diff
@@ -65,7 +73,15 @@ impl<'a> State<'a> {
         self.map.insert(state);
     }
 
-    /// Tires to find an inner map containing T.
+    pub fn transmute<O: Problem>(self: State<'a, P>) -> State<'a, O> {
+        State {
+            parent: self.parent.map(|parent| Box::new(parent.transmute::<O>())),
+            map: self.map,
+            _phantom: std::marker::PhantomData::<O>,
+        }
+    }
```

Comment:
What is this good for?

Reply:
I added some documentation. It's a pretty obscure use-case, but you could reuse `State` for multiple problems this way. This was meant as a replacement because `State` now is generic, and I thought it couldn't hurt.

Reply:
Hmm I'm having a hard time imagining such a usecase. Even reusing the same state for the same problem does not seem particularly useful. Maybe we could remove this for now and only add it once we have a usecase in mind?